rfcs/0002-mvp.md

```diff
@@ -0,0 +1,159 @@
+
+- develop locally
+- push database migrations to their Postgres database
+
+# Motivation
+[motivation]: #motivation
+
+- Make it easy to run Supabase locally
+- Make it easy to manage database migrations
+- Make it easy to do "database branching", where each git branch can have it's own data to prevent conflicts. This is especially important for working with platforms like Vercel/Netlify
+
+
+# Detailed design
+[design]: #detailed-design
+
+
+### Initializing
+
+```sh
+supa init     # creates a .supabase folder with relevant config, connection strings etc. Also scaffolds out a database folder.
+supa start    # Starts Supabase (by running docker-compose), and pulls up the database
+supa eject    # See the .supabase/docker-compose.yml details below
+```
+This will likely need: 
+
+- `supabase.js` - a config file with the Supabase Project details
+- `.supabase/config.txt` - CLI config and details
+- `.supabase/docker-compose.yml` - for running Supabase locally. We may not even need this, as we can just run it directly from the `npm_modules` folder, which would prevent conflicts when we add upgrades or make changes. If we do generate a file, this should be stored in the `.supabase` folder so that it doesn't conflict with the user's docker-compose.yml file. When the user runs `supa eject`, it can move the file out to their root folder (similar to a `cra eject` model), and then the user can manage this themselves.
+
+
+For the database scaffold, [this](https://gitlab.com/delibrium/delibrium-postgrest/-/tree/master/database) format is very nice, although I think we can make it a bit clearer. Here is a suggested format:
+
+```sh
+│   # Data: database fixed/seed data.
+├── data
+│   ├── some_data.{sql|csv|json}        
+│   # Models: Tables and Views
+├── models
+│   ├── public.table                    # Personal preference to explicitly add the schema, and we should prefer dot notation over nesting.
+│   │   ├── data.{sql|csv|json}         # For convenience, table specific data can be stored here too.   
+│   │   ├── model.sql                   # This is the core. We should also figure out a nice, readable format for SQL files. Should include constraints and indexes and table/column comments.
+│   │   ├── permissions.sql             # Grants and Policies.
+│   │   ├── test.sql                    # PGTAP tests?
+│   │   └── triggers.sql                # table triggers 
+│   # Functions/RPC: functions. At the moment the function are written in SQL. If we want to support something like plv8, we would have to include a builder (like webpack) to bundle the function.
+├── functions                                
+│   ├── public.func1.sql                
+│   ├── public.func2.sql│   
+│   # Postgres roles and grants.
+├── roles                                
+│   ├── role_name.sql    
+│   # Types: custom database types
+└── types
+    ├── _init.sql
+    ├── public.some_type.sql
+    └── extensions.some_type.sql
```

Comment:
These seems to be composed of really short files: https://github.com/Delibrium/delibrium-postgrest/tree/master/database/types.

Perhaps we could have a single `types.sql` by default instead. Is there an advantage for migrations that comes with using single files?

I'd be fine either way, but it's more jumping around files to find something when having many tiny ones.